### PR TITLE
Add api handlers for cache and improve caching

### PR DESCRIPTION
**Pull request by JustRoxy**

### CODE REVIEW

**Comment by JustRoxy** [comment]
english softbrain moment, that should mean this thing

File: backend/CircleForms/Models/Posts/PostFields/PostField.cs [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-namespace CircleForms.Models.PostFields;
+namespace CircleForms.Models.Posts.PostFields;
 
 public class PostField
 {
     protected PostField() { }
     public virtual FieldType Type { get; set; }
 
-    public string Title { get; set; }
+    public string Value { get; set; }
```
Comment (stanriders) [inline backend/CircleForms/Models/Posts/PostFields/PostField.cs:10]: Why `value`? It was supposed to be a title of a field, for example "how much pp was cookiezi's blue zenith score worth?"

File: backend/CircleForms/Controllers/PostsController.cs [hunk 2]
```diff
@@ -40,13 +30,47 @@ public async Task<IActionResult> Post(Post post)
...
         }
 
         _logger.LogWarning("User had an invalid name claim: {Claim}", claim);
 
         return BadRequest();
     }
+
+    #region Mongo
+    [HttpGet("/posts/mongo/{id}")]
+    public async Task<Post> Get(string id)
```
Comment (stanriders) [inline backend/CircleForms/Controllers/PostsController.cs:45]: Do we even need to expose that? And why not use a query param on a regular get, for example `/posts/1?cached=false`?

Comment (stanriders) [reply backend/CircleForms/Controllers/PostsController.cs:45]: Keep in mind it's gonna be public, so anyone would be able to use uncached endpoint and potentially overload the database

**Review by stanriders — CHANGES_REQUESTED** [review]

File: backend/CircleForms/Controllers/PostsController.cs (hunk 2 above)
Comment (JustRoxy) [reply backend/CircleForms/Controllers/PostsController.cs:45]: We are not exposing it, see #35

File: backend/CircleForms/Controllers/PostsController.cs [hunk 3]
```diff
@@ -40,13 +30,50 @@ public async Task<IActionResult> Post(Post post)
         if (!string.IsNullOrEmpty(claim) && long.TryParse(claim, out var userId))
         {
             post.AuthorId = userId;
-            var result = await _postRepository.AddPost(userId, post);
+            var result = await _postRepository.Add(userId, post);
 
-            return Ok(result);
+            return CreatedAtAction("Get", new {id = post.Id.ToString()}, result);
```
Comment (stanriders) [inline backend/CircleForms/Controllers/PostsController.cs:35]: `GetCachedPost`? `Get` is admin-only

File: backend/CircleForms/Controllers/UsersController.cs [hunk 4]
```diff
@@ -32,8 +34,26 @@ public async Task<IActionResult> Get(long id)
...
+    {
+        var user = await _usersService.Get(id);
+        user.Roles = (Roles) role;
+        await _usersService.Update(id, user);
+
+        return user;
+    }
+
+    [Authorize]
+    [HttpGet("/Self")]
```
Comment (stanriders) [inline backend/CircleForms/Controllers/UsersController.cs:56]: I'd prefer `/me`, similar to osu api naming

File: backend/CircleForms/Services/Database/PostRepository.cs [hunk 5]
```diff
@@ -23,33 +26,46 @@ public PostRepository(ILogger<PostRepository> logger, IConnectionMultiplexer red
...
-    public async Task<User> AddPost(long id, Post post)
+    public async Task<Post> Add(long id, Post post)
     {
+        post.PublishTime = DateTime.UtcNow;
+
+        var publishUnixTime = ((DateTimeOffset) post.PublishTime).ToUnixTimeMilliseconds();
+
         var update = Builders<User>.Update.AddToSet(x => x.Posts, post);
-        var user = await _users.FindOneAndUpdateAsync(x => x.Id == id, update);
+        await _users.FindOneAndUpdateAsync(x => x.Id == id, update);
```
Comment (stanriders) [inline backend/CircleForms/Services/Database/PostRepository.cs:36]: Maybe confirm it was added before continuing?

**Review by stanriders — CHANGES_REQUESTED** [review]

**Review by stanriders — APPROVED** [review]

**Pull request merged**